WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -594,9 +594,14 @@ private extension OrderDetailsViewController {
     }
 
     private func collectPaymentForCurrentOrder() {
+        let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
+        let currencyCode = ServiceLocator.currencySettings.currencyCode
```

Comment:
Orders can have different currency than the store. Admittedly they don't usually, but they can. Please open an issue to use the currency from the order.

https://github.com/woocommerce/woocommerce/blob/b19500728b4b292562afb65eb3a0c0f50d5859de/includes/abstracts/abstract-wc-order.php#L279-L287

BTW, separate but related -- for card-present payments -- we may eventually need to use the PaymentGatewayAccount supportedCurrencies property to determine whether or not we can collect a payment. I've opened issue #4425 for that.